django/forms/boundfield.py

```diff
@@ -295,11 +295,17 @@ def build_widget_attrs(self, attrs, widget=None):
         if (
             not attrs.get("aria-describedby")
             and not widget.attrs.get("aria-describedby")
-            and self.field.help_text
             and not self.use_fieldset
             and self.auto_id
+            and not self.is_hidden  # Thoughts?
```

Comment:
I think that we should not add `aria-describedby` when the field is hidden. 

Adding help_text to a hidden input I think we could consider a user error. However, hidden inputs could well have validation errors. I don't think we want to associate those to the input given there's no way for the input to be edited. 

These errors are usually shown at the top of the form instead. I'm not sure how they should be announced but it seems to me like that should be a separate PR / change if needed.

Reply:
jj

Reply:
This change looks like a separate cleanup that we could do in advance. As far as I'm aware, it won't make any difference since all `hidden` elements are not visible for both the accessibility tools and browsers presentation layer. We will only reduce the size of the generated HTML.

Reply:
I agree that this guard adding should be done as an orthogonal change to this PR.

Reply:
Merged in c187f5f9242b681abaa199173e02066997439425